synapse/storage/databases/main/event_federation.py

```diff
@@ -1318,16 +1318,14 @@ def _get_missing_events(
 
         query = (
             "SELECT prev_event_id FROM event_edges "
-            "WHERE room_id = ? AND event_id = ? AND is_state = ? "
+            "WHERE event_id = ? AND is_state = ? "
```

Comment:
Since we'll no longer be populating `room_id`, if someone upgrades to Synapse 1.61, then rolls back to 1.60, `/get_missing_events` is going to return an empty list for any events that were sent during the period that 1.61 was running.

My feeling is that that's enough of an edge case that it's not worth delaying all this stuff for another release, but lmk if you disagree.

Reply:
My concern is that we require servers to serve up any prev events of an event its sending out, and it naively looks to me like Synapse is using `/get_missing_events` for that:

https://github.com/matrix-org/synapse/blob/cc945716b2660a0efcc9f659c949352eb9dde4ab/synapse/handlers/federation_event.py#L221-L275

Reply:
I'm not following you, I'm afraid. Yes, we call `/get_missing_events` to fetch missing `prev_events` for any events we receive. What of it?

Reply:
A server will drop events if the call to `get_missing_events` doesn't return the prev events, which seems bad?

Reply:
Yes, my point is that it's rather unlikely, and will soon resolve itself.

In particular, I think the only time it can be a problem is if we failed to send out some event X which was created we were running 1.61. Then, when we downgrade, and attempt federation catchup, X will get dropped, because `get_missing_events` is returning invalid results. In short, a federation transmission that was failing anyway is failing for a bit longer.

Then, as soon as someone sends another event Y in the room, X will get pulled in via a `get_missing_events` on Y, and everything is happy again.

Reply:
Ok, it wasn't clear that you had considered that it would temporarily cause dropped events.

I am not particularly comfortable with us introducing this failure mode, events not being sent in a timely manner is really not ideal. My vote would be to continue inserting `room_id` and remove it after the next RC has been cut, I don't see much disadvantage that would justify introducing this edge case. We should make a PR after the next release to remove those unused rows anyway.

> In particular, I think the only time it can be a problem is if we failed to send out some event X which was created we were running 1.61. Then, when we downgrade, and attempt federation catchup, X will get dropped, because `get_missing_events` is returning invalid results. In short, a federation transmission that was failing anyway is failing for a bit longer.

We send out events asynchronously over federation, so any events that have just been persisted when the server restarts will also likely be hit by this.

Reply:
hmph. *fine*.

Reply:
Ok, I've pulled out #12914. We can land this PR in the following release.